lib/bloc/podcast_clip/podcast_clip_bloc.dart

```diff
@@ -0,0 +1,238 @@
+import 'dart:async';
+import 'dart:developer';
+import 'dart:io';
+import 'package:anytime/services/audio/audio_player_service.dart';
+import 'package:breez/bloc/async_actions_handler.dart';
+import 'package:breez/bloc/podcast_clip/podcast_clip_details_model.dart';
+import 'package:ffmpeg_kit_flutter_https_gpl/ffmpeg_kit.dart';
+import 'package:ffmpeg_kit_flutter_https_gpl/ffmpeg_session.dart';
+
+import 'package:ffmpeg_kit_flutter_https_gpl/return_code.dart';
+import 'package:path_provider/path_provider.dart';
+import 'package:rxdart/rxdart.dart';
+
+enum PodcastClipState {
+  IDLE,
+  FETCHING_IMAGE,
+  FETCHING_AUDIO_CLIP,
+  GENERATING_CLIP
+}
+
+const int _initialSeconds = 10;
+const int _maxClipDuration = 120;
+
+class PodcastClipBloc with AsyncActionsHandler {
+  final BehaviorSubject<PodcastClipDetailsModel> _clipDetailsBehaviourSubject =
+      BehaviorSubject();
+
+  Stream<PodcastClipDetailsModel> get clipDetails =>
+      _clipDetailsBehaviourSubject.stream;
+
+  final BehaviorSubject<PodcastClipState> _clipStateBehaviourSubject =
```

Comment:
Can we hold the status on the PodcastClipState model? It seems a fit and we can reduce the number of streams to one.

Reply:
Done. Thanks